packages/web3-eth/src/eth_tx.ts

```diff
@@ -0,0 +1,315 @@
+		maxPriorityFeePerGas: convertToValidType(transaction.maxPriorityFeePerGas, desiredType),
+		nonce: convertToValidType(transaction.nonce, desiredType),
+		chainId: convertToValidType(transaction.chainId, desiredType),
+		gasLimit: convertToValidType(transaction.gasLimit, desiredType),
+		v: convertToValidType(transaction.v, desiredType),
+		common: {
+			...transaction.common,
+			customChain: {
+				...transaction.common?.customChain,
+				networkId: convertToValidType(
+					transaction.common?.customChain?.networkId,
+					desiredType,
+				),
+				chainId: convertToValidType(transaction.common?.customChain?.chainId, desiredType),
+			},
+		},
+	};
+	// TODO - TSC is complaining that ReturnType could be instantiated with an
+	// arbitrary type which could be unrelated to 'string | number | bigint | undefined'
+	return formattedTransaction as unknown as Transaction<ReturnType>;
+}
+
+export const detectTransactionType = (
+	transaction: Transaction,
+	overrideMethod?: (transaction: Transaction) => Numbers | undefined,
+): Numbers | undefined => {
+	if (overrideMethod !== undefined) return overrideMethod(transaction);
+	if (transaction.type !== undefined) return transaction.type;
+
+	if (transaction.maxFeePerGas !== undefined || transaction.maxPriorityFeePerGas !== undefined)
+		return '0x2';
+	if (transaction.hardfork === 'london') return '0x2';
+	if (transaction.common?.hardfork === 'london') return '0x2';
+
+	if (transaction.accessList !== undefined) return '0x1';
+	if (transaction.hardfork === 'berlin') return '0x1';
+	if (transaction.common?.hardfork === 'berlin') return '0x1';
+
+	return undefined;
+};
+
+const validateCustomChainInfo = (transaction: Transaction) => {
+	if (transaction.common !== undefined) {
+		if (transaction.common.customChain === undefined) throw new MissingCustomChainError();
+		if (transaction.common.customChain.chainId === undefined)
+			throw new MissingCustomChainIdError();
+		if (
+			transaction.chainId !== undefined &&
+			transaction.chainId !== transaction.common.customChain.chainId
+		)
+			throw new ChainIdMismatchError({
+				txChainId: transaction.chainId,
+				customChainId: transaction.common.customChain.chainId,
+			});
+	}
+};
+
+const validateChainInfo = (transaction: Transaction) => {
+	if (
+		transaction.common !== undefined &&
+		transaction.chain !== undefined &&
+		transaction.hardfork !== undefined
+	)
+		throw new CommonOrChainAndHardforkError();
+	if (
+		(transaction.chain !== undefined && transaction.hardfork === undefined) ||
+		(transaction.hardfork !== undefined && transaction.chain === undefined)
+	)
+		throw new MissingChainOrHardforkError({
+			chain: transaction.chain,
+			hardfork: transaction.hardfork,
+		});
+};
+
+const validateGas = (transaction: Transaction<HexString>) => {
+	if (
+		transaction.gas === undefined &&
+		transaction.gasLimit === undefined &&
+		transaction.maxPriorityFeePerGas === undefined &&
+		transaction.maxFeePerGas === undefined
```

Comment:
I prefer to destruct all such params into in condition into boolean identifiers expressions and then user them to construct complex conditions. That make it easy to read and help simplify conditions. 

```ts
const isGasPresent = transaction.gas === undefined && transaction.gasLimit === undefined;
const isMaxPriorityFeePresent = transaction.maxPriorityFeePerGas === undefined && transaction.maxFeePerGas === undefined;


if (isGasPresent && isMaxPriorityFeePresent) {
   hrow new MissingGasError({
}
```